### PR TITLE
Set "SKIP_INSTALL" to "YES" for library projects.

### PR DESCRIPTION
**Pull request by rz-robsn**
Set SKIP_INSTALL to YES for all library projects created through buck except for app targets. Before this change attempting to archive the app would create a generic xcode archive which could not be uploaded to the app store. This config change fixes it (got the solution from this [SO post](https://stackoverflow.com/questions/10715211/cannot-generate-ios-app-archive-in-xcode))

Tested by checking that the lib project Skip Install values are set to Yes after project generation.

@dfed @bachand @fdiaz

### CODE REVIEW

File: Config/configs.bzl [hunk 1]
```diff
@@ -30,11 +30,16 @@ def bundle_identifier(name):
 def library_configs():
     lib_specific_config = {
         "SWIFT_WHOLE_MODULE_OPTIMIZATION": "YES",
+
+        # Setting SKIP_INSTALL to NO for library configs would create
```
[the remark this answers is not in the document]
Comment (rz-robsn) [reply Config/configs.bzl]: @dfed 

> does “library” imply “static?”

Yeah, given the technical note it should mean static. I'll update the comment to specify that.

> is this true for libraries in general or just static libraries?
> if someone were to use this config on a dynamic framework, would that cause a failure? Apple’s documentation doesn’t yield a clear answer on this one

That's a good question for which I don't know the answer to :-p. I guess we should try it out and see what happens.

File: Config/configs.bzl [hunk 2]
```diff
@@ -30,11 +30,16 @@ def bundle_identifier(name):
...
+        # create a generic xcode archive which can not be uploaded the app store
+        "SKIP_INSTALL": "YES",
     }
     library_config = SHARED_CONFIGS + lib_specific_config
     configs = {
         "Debug": library_config,
         "Profile": library_config,
+        "Release": lib_specific_config,
```
[the remark this answers is not in the document]
Comment (bachand) [reply Config/configs.bzl]: This project probably dates back before we settled on just `Debug` and `Release` internally. I suspect we can remove `Profile` now from this project. We should use `Release` for that I think.

Doesn’t have to be this PR

[the remark this answers is not in the document]
Comment (bachand) [reply Config/configs.bzl]: Shouldn’t this line be `library_config` though and not `lib_specific_config`?

File: Config/configs.bzl [hunk 3]
```diff
@@ -30,11 +30,17 @@ def bundle_identifier(name):
 def library_configs():
     lib_specific_config = {
         "SWIFT_WHOLE_MODULE_OPTIMIZATION": "YES",
+
+        # Setting SKIP_INSTALL to NO for static library configs would create
+        # create a generic xcode archive which can not be uploaded the app store
+        # https://developer.apple.com/library/archive/technotes/tn2215/_index.html
+        "SKIP_INSTALL": "YES",
```
Comment (bachand) [inline Config/configs.bzl]: Do we do this in our main app @rz-robsn?

Comment (rz-robsn) [reply Config/configs.bzl]: @bachand Yup, we do this in PR no 59608 (not sure if we can link to our app repo from public airbnb repo so I'm just posting the #).

File: Config/configs.bzl (hunk 2 above)
[the remark this answers is not in the document]
Comment (rz-robsn) [reply Config/configs.bzl]: @bachand `library_config` contains more settings that I'm not sure we want to include for release (like ONLY_ACTIVE_ARCH). We use `lib_specific_config` instead to reduce the scope of this PR.

File: Config/configs.bzl (hunk 1 above)
[the remark this answers is not in the document]
Comment (dfed) [reply Config/configs.bzl]: So, I tried out `SKIP_INSTALL = NO` for dynamic libraries, and the archive wasn't correct. I think the fix you've proposed is right!

**Pull request merged**